### PR TITLE
Add custom_metadata keyword to upload method signature to match Rails 7 requirements

### PR DESCRIPTION
**Pull request by lortza**
## TL;DR
Fixes a bug in uploading images for Rails 7.

## Explanation
I recently updated one of my apps from Rails 6.1.7 to Rails 7.0.4.3 and in doing so, broke my ability to upload images to Dropbox via ActiveStorage. I was able to trace the error back to the upload method here in this gem:

```
ArgumentError: unknown keyword: :custom_metadata
from /Users/myusername/.rvm/gems/ruby-3.1.0/gems/activestorage-dropbox-1.0.0/lib/active_storage/service/dropbox_service.rb:13:in `upload'
```

That lead me to believe that somewhere Rails is expecting this method to have that argument. After doing some digging in the [ActiveStorage source code](https://github.com/rails/rails/search?q=custom_metadata), I noticed that the Rails-supported uploader services have a keyword argument `:custom_metadata` in their method signatures that conveniently matches the one in the error output. 

```ruby
# activestorage/lib/active_storage/service/s3_service.rb
def upload(key, io, checksum: nil, filename: nil, content_type: nil, disposition: nil, custom_metadata: {}, **)

# activestorage/lib/active_storage/service/gcs_service.rb
def upload(key, io, checksum: nil, content_type: nil, disposition: nil, filename: nil, custom_metadata: {})

# activestorage/lib/active_storage/service/azure_storage_service.rb
def upload(key, io, checksum: nil, filename: nil, content_type: nil, disposition: nil, custom_metadata: {}, **)
```

Adding this argument to the local copy of the gem on my machine seems to have fixed the glitch. 

## Testing 
1. in `development.rb`, change `config.active_storage.service` from `:local` to :dropbox`
1. add this argument to my local copy of the gem
1. restart my localhost:3000 server
1. step through the process that originally revealed the bug -- this time, the upload worked
1. check my dropbox folder to ensure the image was there and it was

## Code Choices
* I did not bump the version number of the gem because I saw no trend of that in previous pull requests.
* I did not add tests for this because there were no existing tests for this method.
* I reordered the args to match the Rails-supported services' signatures -- which is not important, just a preference.

### CODE REVIEW

**Comment by ashishprajapati** [comment]
> @ashishprajapati is there anything I can do to help get this PR into production?

Hi @lortza - Nice work and thanks for the follow-up. I have reviewed the PR and will release it to production.

**Review by ashishprajapati — APPROVED** [review]
Looks good!

**Comment by lortza** [comment]
@ashishprajapati I'm still getting that `ArgumentError: unknown keyword: :custom_metadata` error -- but only on my Rails 7 feature branch in production (Heroku). I'm wondering if the problem is that we didn't bump the gem's version number and perhaps Heroku is caching old gems. Since I deleted my local gem and reinstalled it, I can't reproduce this error locally. Have you encountered this sort of problem before with gem versioning?

I should mention that I did not do these steps from the README since it was not a new version number for the gem:
> To install this gem onto your local machine, run `bundle exec rake install`. To release a new version, update the version number in `version.rb`, and then run `bundle exec rake release`, which will create a git tag for the version, push git commits and tags, and push the `.gem` file to [rubygems.org](https://rubygems.org/).

**Comment by ashishprajapati** [comment]
> @ashishprajapati I'm still getting that `ArgumentError: unknown keyword: :custom_metadata` error -- but only on my Rails 7 feature branch in production (Heroku). I'm wondering if the problem is that we didn't bump the gem's version number and perhaps Heroku is caching old gems. Since I deleted my local gem and reinstalled it, I can't reproduce this error locally. Have you encountered this sort of problem before with gem versioning?

Hey @lortza - we've already bumped the gem version to 2.0.0, which you can find on RubyGems here: https://rubygems.org/gems/activestorage-dropbox/versions/2.0.0. It's working well and it's important to note that bumping the gem version is mandatory when releasing updates.

It seems like the issue might be related to Heroku not using the correct version of the gem:
- Make sure you've specified the correct version of the activestorage-dropbox gem in your Gemfile, i.e., `gem 'activestorage-dropbox', '~> 2.0'`. Update the version number to 2.0 if necessary and run bundle update activestorage-dropbox in your terminal to update the gem in your Gemfile.lock.
- Ensure your production environment is using the correct gem versions. Check your Gemfile.lock to see if there's any conflict in the gem dependencies.

AP

**Comment by lortza** [comment]
@ashishprajapati that makes all the sense. After a regular bundle, my Gemfile.lock was still stuck on version `1.0.0`. I wasn't sure what the latest published version was supposed to be. Annnd I probably should have looked https://rubygems.org/gems/activestorage-dropbox/versions/2.0.0. 🤦 🤣  Anyway, I've updated my gem to version `2.0.0` and everything is working in prod. Thanks for publishing this fix. I'm glad I can up upgrade my app to Rails 7 now!
